x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/planner/ExpressionTranslators.java

```diff
@@ -284,6 +285,9 @@ static Query translate(BinaryComparison bc, TranslatorHandler handler) {
             ZoneId zoneId = null;
             if (DataType.isDateTime(attribute.dataType())) {
                 zoneId = bc.zoneId();
+                value = DEFAULT_DATE_TIME_FORMATTER.formatMillis((Long) value);
+                format = DEFAULT_DATE_TIME_FORMATTER.pattern();
+                isDateLiteralComparison = true;
```

Comment:
Checked myself, for some reason both are necessary to make the tests pass :/

Reply:
Yeah, it does look confusing. The change in `ExpressionTranslators` in esql-core is for the test of IN list predicate on datetime fields. If the fix for equal predicate is valid, the `InComparisons` can be moved into `EsqlExpressionTranslators`, so that these two places can be consolidated into one.